kong/clustering/rpc/socket.lua

```diff
@@ -59,12 +59,37 @@ function _M:_get_next_id()
 end
 
 
+function _M:_report_error(payload_id, code, err)
+  local res, rpc_err = self.outgoing:push(
+    new_error(payload_id, code,  err)
+  )
+
+  if not res then
+    ngx_log(ngx_WARN, "[rpc] unable to push RPC call error: ", rpc_err)
+  end
+end
+
+
 function _M._dispatch(premature, self, cb, payload)
   if premature then
     return
   end
 
-  local res, err = cb(self.node_id, unpack(payload.params))
+  local res, except, err, trace
```

Comment:
Could we use the return values of xpcall, like https://github.com/Kong/kong/blob/master/kong/clustering/config_helper.lua#L368

Reply:
It may be
```
local ok, res_or_err = xpcall(cb, function(err) ... end, self.node_id, unpack(payload.params))
```

Reply:
I prefer this method. It leaves us more freedom to manipulate the output of the error